testing/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractCrossVersionPerformanceTest.groovy

```diff
@@ -70,6 +70,8 @@ class AbstractCrossVersionPerformanceTest extends AbstractPerformanceTest {
         runner.addBuildMutator { invocationSettings ->
             new ApplyGradleEnterprisePluginMutator(invocationSettings.projectDir)
         }
+        // Required by GradleLifecycle API used by Develocity
+        runner.gradleOpts += ["--add-opens=java.base/java.lang.invoke=ALL-UNNAMED"]
```

Comment:
🤔 Do we add this add-opens automatically in other places, like when using the TAPI or the wrapper? Why doesn't that happen automatically for the performance tests?

Reply:
AFAICS the `add-opens` are added to tests that require it, such as the tests that explicitly enable the configuration cache. 

`GradleLifecycle` introduces the dependency on Configuration Cache serialization.

`Develocity` relies on `GradleLifecycle` now so we know the `add-opens` will be required here.

In any case, I see this as a quick fix to unblock the release. I'm glad to let devprod consider what the best strategy should be long term.

Reply:
And as I said in the Slack thread, this is not new, we do that here, for example, https://github.com/gradle/gradle/blob/1a5e48701b602def428d9c79e3793adff7a4f798/testing/performance/src/performanceTest/groovy/org/gradle/performance/regression/java/JavaConfigurationCachePerformanceTest.groovy#L59

Reply:
I want to make sure that this is only a problem for our tests and not in production. Though I see now that we already sprinkled `--add-opens` in quite a few places in our code base. So I suppose it is fine.

Reply:
The Gradle client adds the required `--add-opens` when starting the daemon, I don't know why but it seems the performance tests are bypassing the client.